mitodl/ocw-to-hugo
# Update and fix link processing

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes #159 
Fixes #173 

#### What's this PR do?

 - Removes use of `getpage` shortcode and uses relative links instead. All links should be functional to my knowledge, let me know if you see an edge case where a link doesn't look right.
 - `pathToChildRecursive` and `getHugoPathSuffix` are removed. Instead, paths are generated in the beginning all at once in `buildPaths` and `buildPathRecursive`.

Most of the changed lines are whitespace due to indentation changes in tests, so you may want to add a `?w=1` to remove whitespace: https://github.com/mitodl/ocw-to-hugo/pull/175/files?w=1

#### How should this be manually tested?
Convert all courses on master, then convert all courses with this branch, then compare the two directories with a file comparison tool. Use `--verbose` to show any errors logged in the output. There will be many differences but in general you should see `{{ getpage ... }}` shortcodes converted to simple relative links, and those links should start `/section` instead of `/course/course-slug-goes-here/section/...`. You should be able to convert a course

Also, check that the issues are fixed:
 - #159 - PDF links should work on `16-01-unified-engineering-i-ii-iii-iv-fall-2005-spring-2006` at `/sections/thermo-propulsion`
 -  #173 - PDF links should work on `18-01-single-variable-calculus-fall-2006 ` at `/sections/readings/course-reader/`

Known issues:
 - There are many warnings about a missing uid `42b179986219b2af6276e22208e1479f` in `6-050j-information-and-entropy-spring-2008` because it's used as a `parent_uid` but there is no other reference. This appears in the plone JSON files too so I don't see what we can do here.
 - There are a bunch of other warnings about missing uids which could be fixed in ocw-data-parser. I made a couple issues: https://github.com/mitodl/ocw-data-parser/issues/123 and https://github.com/mitodl/ocw-data-parser/issues/124
